### ISSUE TITLE
setting additionalProperties to true

### ISSUE THREAD

**Issue by sramam** [vega/typescript-to-json-schema#9]
Not sure if this belongs in the vega fork or the original repo.
Looking at the code, it's not clear that 'additionalProperties' can ever be set to true in the generated schema. 

Best I could figure was this function https://github.com/vega/typescript-to-json-schema/blob/master/src/NodeParser/InterfaceNodeParser.ts#L65 and it'll always return false. 

There seem to be no test cases to the counter. 

Would appreciate clarification and/or some idea of how to go about fixing this. Happy to submit a PR once given a scent to follow.

**Comment by domoritz** [comment]
Usually, you wouldn't set it to true but define what types you expect additional properties to be. This is useful for mapsas in https://github.com/vega/typescript-to-json-schema/blob/master/test/valid-data/type-maps/schema.json#L38. Does this make sense?

**Comment by sramam** [comment]
ah! 
Two questions: 

1.  Is there a way to specify this in the TypeScript definition? Ideally, @additionalProperties annotation on an object.
2. Currently, this is being set to false. Which seems to trigger a `no additionalProperties allowed` check from a validator I am using. 

Per the spec, http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.10
it's being absent is treated as "empty schema", which should validate arbit additional props.

Basically, providing #1 will solve #2 I think. And current behavior seems to be amiss from the standard.

Or did I miss something else again?! 😱

**Comment by domoritz** [comment]
1. I haven't added anything but I don't see why we wouldn't be able to add it
2. That seems to be the behavior you want from a json schema that is generated from typescript. https://github.com/YousefED/typescript-json-schema has an option to not automatically output `additionalProperties: false` but I haven't bothered adding this to this generator because I always want it. 

I don't agree that what I'm doing is not a according to the standard. What this library does is try to create a correspondence between the behavior in typescript and the behavior in json schema. Can you give me an example where you would want a different behavior?

Note that you can a write in typescript (roughly, I don't have a compiler right now) 

```
interface A {
  foo: number
  [key: string]: any
}
```

which will allow additional properties on the interface A.

**Comment by sramam** [comment]
I see. That intent wasn't fully clear - from a reading of the code.

I now see how current behavior is actually a better approach to
determining when additionalProperties should be included and when not.
It's a more natural extension of excess-property-checking in TypeScript.

Nicely done. Thanks much.

**Comment by yliu-d** [comment]
@domoritz we are using this great library but we found we need to allow customers sending JSON files with additional properties anywhere they want in JSON. Can I create a PR to add a `--allowAdditionalProperties` flag? Thanks!

**Comment by domoritz** [comment]
@yliu-d would it make sense to just iterate over the generated JSON to change the property instead of adding support to the library itself?

**Comment by yliu-d** [comment]
@domoritz that's the workaround we are using today, yes we can continue using this workaround.

**Comment by tamj0rd2** [comment]
Would you accept a PR to add a flag for this?

My use case is that the types we're defining represent a contract, whereby an object only needs to satisfy the constraints of the type.

**Comment by domoritz** [comment]
I'd consider a pull request with tests if the changes to the library code are small (which I expect them to be).

**Comment by tamj0rd2** [comment]
Do you have any guidance on this? Should it be as easy as adding an allowAdditional flag to the different Object parser constructors? e.g InterfaceAndClassNodeParser and ObjectLiteralExpressionNodeParser
